User Story 154682|User Story 154680| Doc customer scenarios (#19300)
microsoftgraph/microsoft-graph-docs: concepts/msgraph-customer-update-assignment-grade-data-for-tool.md

---
title: "Update data in assignments and grades"
description: "Use Microsoft Graph to update data in assignments and grades for reporting, communication, and automation tools."
ms.localizationpriority: medium
author: "AshwaniBansal1"
ms.prod: "education"
doc_type: conceptualPageType
---

> # Update data in assignments and grades

Reviewer: Does this article mention anything about updating the data? The APIs mentioned are about getting information, not updating it.

> Education institutions have custom tools that needs to update assignments and grades data; for example, user needs to create an assignment with custom tool, or update grades data ( outcome) for the submission from students.

Reviewer 2: ```suggestion
Education institutions have custom tools to update assignments and grading data. For example, a user may need to create an assignment or update grading data (outcome) with a custom tool for students to submit them.
```

> **Note:** You can use [Graph Explorer](https://developer.microsoft.com/en-us/graph/graph-explorer) to test the [Education APIs](https://learn.microsoft.com/en-us/graph/msgraph-onboarding-graphexplorer) mentioned in this article.

## Get classes and members information

> All the assignments and grades information is part of the team in Microsoft Teams. You can get information about classes, members, and roles (student or teacher) in a team by using the following Teams APIs:

Reviewer 2: ```suggestion
All the assignments and grading information are part of the team in Microsoft Teams. You can get information about classes, members, and roles (student or teacher) in a team by using the following Teams APIs:
```

- [List classes of an educationSchools](/graph/api/educationschool-list-classes): Returns a list of all classes in your education tenant.
- [List members of an educationClass](/graph/api/educationclass-list-members): Lists information about members of a specific class.

> After you get the relevant class and member information, you can get the assignment and grade information that you need.

Reviewer 2: ```suggestion
After you get the relevant class and member information, you can get the assignment and grade information you need.
```

## Get class assignment information

All assignments and students' information are linked to class and submission information respectively. The following APIs retrieve information about class assignments:

> - [educationAssignment resource type](/graph/api/resources/educationassignment): Lists all data for assignments in a class.

Reviewer: Did you mean to link to List and Get method topics here?

- [educationSubmission resource type](/graph/api/resources/educationsubmission) : A `submissionId` is created for every student for whom an assignment is published. You can use this API to get the submission status - that is, whether it was turned in and graded,([resources](/graph/api/educationsubmissionresource-get)) submitted by the student, and the grade ([outcome](/graph/api/resources/educationoutcome)) for the submission.

## Permissions

> For update of data in assignments user get support for both delegated (per user) and AppOnly (per App). AppOnly support is currently only for Beta version.

Reviewer 2: ```suggestion
Delegated (per user) and AppOnly (per App) permissions are required to update data in assignments. AppOnly permissions are currently supported in beta versions only.
```

> > **Note:** Since Student Information and Grade information is sensitive data, it needs to be verified if AppOnly data is what is needed for the scenario.

Reviewer 2: ```suggestion
> **Note:**  Since student and grade information is sensitive data, please verify which permissions are required. 
```

---
title: "Update data in assignments and grades"

> description: "Use microsoft Graph to update data in assignments and grades for reporting, communication and automation tools."

Reviewer 3: **M**icrosoft

ms.localizationpriority: medium
author: "ashwani-bansal"
ms.prod: "education"
doc_type: conceptualPageType
---

# Update data in assignments and grades

> ## Scenario

Reviewer 3: remove

Education institutions have custom tools that needs to update assignments and grades data; for example, user needs to create an assignment with custom tool, or update grades data ( outcome) for the submission from students.

> **Note:** You can use [Graph Explorer](https://developer.microsoft.com/en-us/graph/graph-explorer) to test the [Education APIs](https://learn.microsoft.com/en-us/graph/msgraph-onboarding-graphexplorer) mentioned in this article.

Reviewer 3: prefix ">" similar to the other scenario

> ## API to use for classes and members information

Reviewer 3: Get classes and members information

- [List classes of an educationSchools](/graph/api/educationschool-list-classes): Returns a list of all classes in your education tenant.
- [List members of an educationClass](/graph/api/educationclass-list-members): Lists information about members of a specific class.

After you get the relevant class and member information, you can get the assignment and grade information that you need.

## Get class assignment information

All assignments and students' information are linked to class and submission information respectively. The following APIs are to retrieve information about class assignments:

- [educationAssignment resource type](/graph/api/resources/educationassignment) : Allows you to list all data for assignments in a class.
- [educationSubmission resource type](/graph/api/resources/educationsubmission) : A `submissionId` is created for every student for whom an assignment is published. You can use this API to get the submission status - that is, whether it was turned in and graded,([resources](/graph/api/educationsubmissionresource-get)) submitted by the student, and the grade ([outcome](/graph/api/resources/educationoutcome)) for the submission.

> ## Permissions required

Reviewer 3: Permissions

For update of data in assignments user get support for both delegated (per user) and AppOnly (per App). AppOnly support is currently only for Beta version.

> **Note:** Since Student Information and Grade information is sensitive data, it needs to be verified if AppOnly data is what is needed for the scenario.

Reviewer 3: prefix ">"

Before you can update data, you must get the relevant class and member information.

> **Note:** You can use [Graph Explorer](https://developer.microsoft.com/en-us/graph/graph-explorer) to test the [Education APIs](https://learn.microsoft.com/en-us/graph/msgraph-onboarding-graphexplorer) mentioned in this article.

## Get class and member information

All the assignments and grading information are part of the team in Microsoft Teams. You can get information about classes, members, and roles (student or teacher) in a team by using the following Teams APIs:

- [List classes of an educationSchools](/graph/api/educationschool-list-classes): Returns a list of all classes in your education tenant.
- [List members of an educationClass](/graph/api/educationclass-list-members): Lists information about members of a specific class.

After you get the relevant class and member information, you can get the assignment and grade information you need.

## Get assignment and submission information

All assignment and student information is linked to class and submission information, respectively. The following APIs retrieve information about class assignments and submissions:

> - [educationAssignment resource type](/graph/api/resources/educationassignment): Lists all the methods available for assignments in a class.

Reviewer: These are not list/get operation topics, they are links to resources. Are you linking to the right topics here?

Author: linking to the education assignments and submissions for the required methods

- [educationSubmission resource type](/graph/api/resources/educationsubmission): A `submissionId` is created for every student for whom an assignment is published. You can use this API to get the submission status - for example, whether it was turned in and graded, ([resources](/graph/api/educationsubmissionresource-get)) submitted by the student, and the grade ([outcome](/graph/api/resources/educationoutcome)) for the submission.

## Update grade (outcome) data

> The outcome data will be recalculated based on changes in the property values. To update student points outcome data, see [update a points outcome](/graph/api/educationoutcome-update?#example-2-update-a-points-outcome).

Reviewer: I'm not sure if these is completely clear. You might want to provide some more detail here about how the values actually change - what is the recalculation based on? 

Per my earlier comment, this topic would also benefit from and end-to-end example rather than just links to other topics.

Author: updated

---
title: "Update assignment and grade data"
description: "Use Microsoft Graph APIs to update assignment and grade data for use in custom tools."
ms.localizationpriority: medium
author: "AshwaniBansal1"
ms.prod: "education"
doc_type: conceptualPageType
---

# Update assignment and grade data

> Education institutions have custom tools to update assignment and grade data. For example, a user might need to create an assignment or update a grade (outcome) with a custom tool for students to submit them. You can use education APIs in Microsoft Graph to update assignments and grades. 

Reviewer: The second sentence is not clear. See if this suggested revision is accurate: "For example, a teacher might need to make an update to an assignment before students submit them, or make a change to a grade (outcome) by using a custom tool."
